tethysext/atcore/tests/integrated_tests/models/files_database/file_collection_client_tests.py

```diff
@@ -463,3 +474,327 @@ def test_bad_collection_client(self):
+        with collection_client.open_file('file1.txt', 'w') as f:
+            file_text = f.write('This text should be written to file.')
+        with open(os.path.join(collection_client.path, 'file1.txt')) as of:
+            file_text = of.read()
+            self.assertEqual('This text should be written to file.', file_text)
+
+    def test_open_file_does_not_exist(self):
+        """Test opening a file that doesn't exist throws the correct error."""
+        test_dir_name = 'test_open_file_does_not_exist'
+        base_files_root_dir = os.path.join(self.test_files_base, test_dir_name)
+        root_dir = os.path.join(self.test_files_base, 'temp', test_dir_name)
+        self.copy_files_to_temp_directory(base_files_root_dir, root_dir)
+        _, collection_instance = self.get_database_and_collection(
+            database_id=self.general_database_id, collection_id=self.general_collection_id,
+            root_directory=root_dir, database_meta={}, collection_meta={}
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        with self.assertRaises(FileCollectionItemNotFoundError) as exc:
+            with collection_client.open_file('file1.txt', 'w') as _:
+                pass
+        self.assertTrue('"file1.txt" not found in this collection.' in str(exc.exception))
+
+    def test_open_file_directory(self):
+        """Test opening a directory throws an error."""
+        test_dir_name = 'test_open_file_directory'
+        base_files_root_dir = os.path.join(self.test_files_base, test_dir_name)
+        root_dir = os.path.join(self.test_files_base, 'temp', test_dir_name)
+        self.copy_files_to_temp_directory(base_files_root_dir, root_dir)
+        _, collection_instance = self.get_database_and_collection(
+            database_id=self.general_database_id, collection_id=self.general_collection_id,
+            root_directory=root_dir, database_meta={}, collection_meta={}
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        with self.assertRaises(IsADirectoryError):
+            with collection_client.open_file('dir1', 'r') as _:
+                pass
+
+    def test_walk(self):
+        """Test walking a collection."""
+        test_dir_name = 'test_walk'
+        base_files_root_dir = os.path.join(self.test_files_base, test_dir_name)
+        root_dir = os.path.join(self.test_files_base, 'temp', test_dir_name)
+        self.copy_files_to_temp_directory(base_files_root_dir, root_dir)
+        _, collection_instance = self.get_database_and_collection(
+            database_id=self.general_database_id, collection_id=self.general_collection_id,
+            root_directory=root_dir, database_meta={}, collection_meta={}
+        )
+        collection_client = FileCollectionClient(self.session, self.general_collection_id)
+        walk_output = [x for x in collection_client.walk()]
+        expected_walk_output = [
```

Comment:
This test will be more effective if you have several levels of nested directories. `walk` should recursively parse into each.

Reply:
Not sure what you meant here. I had three levels. I add a second directory in one of the levels.

Reply:
The hard-coded "." was throwing me off.